src/main/java/com/cloudbees/jenkins/plugins/customtools/CustomTool.java

```diff
@@ -289,9 +310,18 @@ public PathsList invoke(File f, VirtualChannel channel)
                 }
                 
                 // Resolve home dir
-                File homeDir = new File(getHome());   
+                final String toolHome = getHome();
+                if (toolHome == null) {
+                    throw new IOException("Cannot retrieve Tool home directory. Should never happen ant this stage, please file a bug");
+                }
+                final File homeDir = new File(toolHome);
                 return new PathsList(outList, homeDir.getAbsolutePath());               
             };
+
+            @Override
+            public void checkRoles(RoleChecker roleChecker) throws SecurityException {
+                // Do nothing
```

Comment:
:bug: use `MasterToSlaveFileCallable` as this is for code intended to be run on an agent from the master only.